ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/explorer/project/SearchNodeHandler.java

```diff
@@ -0,0 +1,250 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.part.explorer.project;
+
+import com.google.common.base.Strings;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+
+import org.eclipse.che.api.promises.client.Promise;
+import org.eclipse.che.api.promises.client.callback.AsyncPromiseHelper;
+import org.eclipse.che.ide.api.project.node.HasStorablePath;
+import org.eclipse.che.ide.api.project.node.Node;
+import org.eclipse.che.ide.ui.smartTree.Tree;
+import org.eclipse.che.ide.ui.smartTree.event.BeforeExpandNodeEvent;
+import org.eclipse.che.ide.ui.smartTree.event.BeforeExpandNodeEvent.BeforeExpandNodeHandler;
+import org.eclipse.che.ide.ui.smartTree.event.ExpandNodeEvent;
+import org.eclipse.che.ide.ui.smartTree.event.ExpandNodeEvent.ExpandNodeHandler;
+import org.eclipse.che.ide.ui.smartTree.event.NodeAddedEvent;
+import org.eclipse.che.ide.ui.smartTree.event.NodeAddedEvent.NodeAddedEventHandler;
+import org.eclipse.che.ide.ui.smartTree.event.PostLoadEvent;
+import org.eclipse.che.ide.ui.smartTree.event.PostLoadEvent.PostLoadHandler;
+
+import java.util.List;
+
+/**
+ * Search node handler, perform searching specified node in the tree by storable value.
+ * For example if user passes "/project/path/to/file" then this node handler will check
+ * opened root nodes and if it contains project node with path "/project" then it will
+ * search children by path "path/to/file".
+ * @author Vlad Zhukovskiy
```

Comment:
It seems that it is not formatted, empty line required before the <code>@author</code> line.